ui/src/app/applications/components/application-details/application-resource-list.tsx

```diff
@@ -6,88 +6,120 @@ import {ResourceIcon} from '../resource-icon';
 import {ResourceLabel} from '../resource-label';
 import {ComparisonStatusIcon, HealthStatusIcon, nodeKey} from '../utils';
 import {Consumer} from '../../../shared/context';
+import {services} from '../../../shared/services';
 
 export const ApplicationResourceList = ({
     resources,
     onNodeClick,
-    nodeMenu
+    nodeMenu,
+    application
 }: {
-    resources: models.ResourceStatus[];
+    resources: any[];
     onNodeClick?: (fullName: string) => any;
     nodeMenu?: (node: models.ResourceNode) => React.ReactNode;
-}) => (
-    <div className='argo-table-list argo-table-list--clickable'>
-        <div className='argo-table-list__head'>
-            <div className='row'>
-                <div className='columns small-1 xxxlarge-1' />
-                <div className='columns small-2 xxxlarge-2'>NAME</div>
-                <div className='columns small-3 xxxlarge-4'>GROUP/KIND</div>
-                <div className='columns small-3 xxxlarge-3'>NAMESPACE</div>
-                <div className='columns small-2 xxxlarge-2'>STATUS</div>
+    application?: models.Application;
+}) => {
+    const [resourceRows, setResourceRows] = React.useState(resources);
+    React.useEffect(() => {
+        setResourceRows(resources);
+        async function getLiveState() {
+            if (application) {
+                const liveStatePromises = resources.map(async resource => {
+                    const resourceRow: any = {...resource};
+                    const liveState = await services.applications.getResource(application.metadata.name, resource).catch(() => null);
+                    if (liveState?.metadata?.annotations && liveState?.metadata?.annotations['argocd.argoproj.io/hook']) {
+                        resourceRow.syncOrder = liveState?.metadata.annotations['argocd.argoproj.io/hook'];
+
+                        if (liveState?.metadata?.annotations['argocd.argoproj.io/sync-wave']) {
+                            resourceRow.syncOrder = resourceRow.syncOrder + ': ' + liveState?.metadata.annotations['argocd.argoproj.io/sync-wave'];
```

Comment:
Consider moving these annotations to constants that can be reused

Reply:
@terrytangyuan Good suggestion! Code updated, please confirm the change. Thanks!